be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -553,28 +554,63 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
-            COUNTER_SET(prepare_timer, prepare_time);
+            COUNTER_SET(prepare_timer, profiler.prepare_time);
+            auto* prepare_query_ctx_timer = ADD_CHILD_TIMER(fragment_ctx->runtime_state()->runtime_profile(),
+                                                            "prepare-query-ctx", "FragmentInstancePrepareTime");
+            COUNTER_SET(prepare_query_ctx_timer, profiler.prepare_query_ctx_time);
+
+            auto* prepare_fragment_ctx_timer = ADD_CHILD_TIMER(fragment_ctx->runtime_state()->runtime_profile(),
+                                                               "prepare-fragment-ctx", "FragmentInstancePrepareTime");
+            COUNTER_SET(prepare_fragment_ctx_timer, profiler.prepare_fragment_ctx_time);
+
+            auto* prepare_runtime_state_timer = ADD_CHILD_TIMER(fragment_ctx->runtime_state()->runtime_profile(),
+                                                                "prepare-runtime-state", "FragmentInstancePrepareTime");
+            COUNTER_SET(prepare_runtime_state_timer, profiler.prepare_runtime_state_time);
+
+            auto* prepare_pipeline_driver_timer =
+                    ADD_CHILD_TIMER(fragment_ctx->runtime_state()->runtime_profile(), "prepare-pipeline-driver",
+                                    "FragmentInstancePrepareTime");
+            COUNTER_SET(prepare_pipeline_driver_timer, profiler.prepare_runtime_state_time);
         } else {
             _fail_cleanup();
         }
     });
-    SCOPED_RAW_TIMER(&prepare_time);
-    RETURN_IF_ERROR(exec_env->query_pool_mem_tracker()->check_mem_limit("Start execute plan fragment."));
 
-    RETURN_IF_ERROR(_prepare_query_ctx(exec_env, request));
-    RETURN_IF_ERROR(_prepare_fragment_ctx(request));
-    RETURN_IF_ERROR(_prepare_workgroup(request));
-    RETURN_IF_ERROR(_prepare_runtime_state(exec_env, request));
-    RETURN_IF_ERROR(_prepare_exec_plan(exec_env, request));
-    RETURN_IF_ERROR(_prepare_global_dict(request));
-    RETURN_IF_ERROR(_prepare_pipeline_driver(exec_env, request));
-    RETURN_IF_ERROR(_prepare_stream_load_pipe(exec_env, request));
+    SCOPED_RAW_TIMER(&profiler.prepare_time);
+    RETURN_IF_ERROR(exec_env->query_pool_mem_tracker()->check_mem_limit("Start execute plan fragment."));
+    {
+        SCOPED_RAW_TIMER(&profiler.prepare_query_ctx_time);
+        RETURN_IF_ERROR(_prepare_query_ctx(exec_env, request));
+    }
+    {
+        SCOPED_RAW_TIMER(&profiler.prepare_fragment_ctx_time);
+        RETURN_IF_ERROR(_prepare_fragment_ctx(request));
+    }
+    {
+        SCOPED_RAW_TIMER(&profiler.prepare_runtime_state_time);
+        RETURN_IF_ERROR(_prepare_workgroup(request));
+        RETURN_IF_ERROR(_prepare_runtime_state(exec_env, request));
+        RETURN_IF_ERROR(_prepare_exec_plan(exec_env, request));
+        RETURN_IF_ERROR(_prepare_global_dict(request));
+    }
+    {
+        SCOPED_RAW_TIMER(&profiler.prepare_pipeline_driver_time);
```

Comment:
How about do not take these metrics into account if they are pretty low, such as several milliseconds.

For example, `SCOPED_RAW_TIMER(xxx, 10ms)`